Src/ILGPU/IR/Construction/IRBuilder.cs

```diff
@@ -106,6 +106,22 @@ protected IRBuilder(BasicBlock basicBlock)
         public ValueReference CreateLaneIdxValue(Location location) =>
             Append(new LaneIdxValue(GetInitializer(location)));
 
+        /// <summary>
+        /// Creates a node that represents the length of dynamic memory.
+        /// </summary>
+        /// <param name="location">The current location.</param>
+        /// <param name="elementType">The element type.</param>
+        /// <param name="addressSpace">The target address space.</param>
+        /// <returns>A reference to the requested value.</returns>
+        public ValueReference CreateDynamicMemoryLengthValue(
```

Comment:
Proposal: Maybe we should make this function `private` to the `IRBuilder` to ensure that it will be used in the scope of a dynamic allocation only 🤓 What do you think?

Reply:
Fixed.